src/analyzer/protocol/ssl/dtls-protocol.pac

```diff
@@ -83,4 +118,11 @@ refine connection SSL_Conn += {
 		}
 		%}
 
+		function choose_record_type(firstbyte: uint8): bool
+			%{
+			uint16_t negotiated_version = zeek_analyzer()->GetNegotiatedVersion();
+			if ( negotiated_version == DTLSv13 && ( (firstbyte & 0x20) == 0x20 ) )
```

Comment:
This is testing the "first three high bits" being `001`, correct? Seems it needs to `&` with a mask of the first 3 bits `0xE0` and compare that result with `0x20` thereby asserting that the first two bits are actually `0`?
```suggestion
			return negotiated_version == DTLSv13 && ( (firstbyte & 0xE0) == 0x20 );
```

With the existing test `111` would still evaluate to true as the first 2 bits are ignored out.

Other nit, given this returns `bool` if the name was `is_unified_record()` that would help to know what the meaning is without reading a comment or the code.

Reply:
you are right, I am not entirely sure what I was thinking here. Thank you for noticing that.